src/Gtk/Avalonia.Gtk3/WindowImpl.cs

```diff
@@ -9,8 +9,27 @@ namespace Avalonia.Gtk3
 {
     class WindowImpl : WindowBaseImpl, IWindowImpl
     {
+        private WindowState _lastWindowState;
+
         public WindowImpl() : base(Native.GtkWindowNew(GtkWindowType.TopLevel))
         {
+            ConnectEvent("window-state-event", OnWindowStateChanged);
+        }
+
+        private unsafe bool OnWindowStateChanged(IntPtr gtkwidget, IntPtr ev, IntPtr userdata)
+        {
+            var windowStateEvent = (GdkEventWindowState*)ev;
+            var newWindowState = windowStateEvent->new_window_state;
+            var windowState = newWindowState.HasFlag(GdkWindowState.Maximized) ? WindowState.Maximized
```

Comment:
Iconified should take precedence over Maximized

Reply:
`IsMiniaturized` in MonoMac should take precendence over `IsZoomed` too?

Reply:
Yes, I guess so. It doesn't matter if window is maximized or not, if it's not visible to the user anyway.

I'm also considering to change the API and make those two separate properties.